data/src/main/java/com/linkedin/data/schema/resolver/ClasspathResourceDataSchemaResolver.java

```diff
@@ -33,61 +34,106 @@
+  /**
+   * Construct a new instance that uses the specified {@link ClassLoader}.
+   *
+   * @deprecated The parserFactory is not needed as this class now uses builtin parsers. Use
+   * {@link #ClasspathResourceDataSchemaResolver(ClassLoader)} instead
+   * @param classLoader provides the {@link ClassLoader}.
+   */
+  @Deprecated
+  public ClasspathResourceDataSchemaResolver(SchemaParserFactory parserFactory, ClassLoader classLoader)
   {
-    return InternalConstants.PEGASUS_DIR_IN_JAR + "/" + schemaName.replace('.', '/') + DEFAULT_EXTENSION;
+    this(classLoader);
   }
 
-  @Override
-  protected NamedDataSchema locateDataSchema(String schemaName, StringBuilder errorMessageBuilder)
+  private class SingleFormatClasspathSchemaResolver extends DefaultDataSchemaResolver
   {
-    NamedDataSchema schema = null;
-    final String schemaResourcePath = getDataSchemaResourcePath(schemaName);
-    try (InputStream stream = _classLoader.getResourceAsStream(schemaResourcePath))
+    /**
+     * Construct a new instance that uses the {@link Thread#getContextClassLoader()} for the current thread.
+     */
+    public SingleFormatClasspathSchemaResolver(DataSchemaParserFactory parserFactory)
+    {
+      super(parserFactory);
+    }
+
+    private String getDataSchemaResourcePath(String schemaName)
     {
-      if (stream == null)
+      return InternalConstants.PEGASUS_DIR_IN_JAR + "/" + schemaName.replace('.', '/') + _extension;
+    }
+
+    @Override
+    protected NamedDataSchema locateDataSchema(String schemaName, StringBuilder errorMessageBuilder)
+    {
+      NamedDataSchema schema = null;
+      final String schemaResourcePath = getDataSchemaResourcePath(schemaName);
+      try (InputStream stream = _classLoader.getResourceAsStream(schemaResourcePath))
       {
-        errorMessageBuilder.append(String.format("Unable to find data schema file \"%s\" in classpath.", schemaResourcePath));
+        if (stream != null)
+        {
+          DataSchemaLocation location = new FileDataSchemaLocation(new File(schemaResourcePath));
+          schema = parse(stream, location, schemaName, errorMessageBuilder);
+        }
       }
-      else
+      catch (IOException e)
       {
-        DataSchemaLocation location = new FileDataSchemaLocation(new File(schemaResourcePath));
-        schema = parse(stream, location, schemaName, errorMessageBuilder);
+        errorMessageBuilder.append(String.format("Failed to read/close data schema file \"%s\" in classpath: \"%s\"", schemaResourcePath, e.getMessage()));
       }
+      return schema;
     }
-    catch (IOException e)
+
+    void setExtension(String extension)
     {
-      errorMessageBuilder.append(String.format("Failed to read/close data schema file \"%s\" in classpath: \"%s\"", schemaResourcePath, e.getMessage()));
+      this._extension = extension;
     }
-    return schema;
-  }
 
-  private final ClassLoader _classLoader;
+    private String _extension = SchemaParser.FILE_EXTENSION;
```

Comment:
I may prefer that we make this `final`, and initialized in constructor and hide `setter` for this.